src/EditorFeatures/VisualBasicTest/SplitOrMergeIfStatements/MergeNestedIfStatementsTests.vb

```diff
@@ -0,0 +1,1394 @@
+            [||]if b then
+                System.Console.WriteLine(a andalso b)
+            else
+                System.Console.WriteLine(a)
+            end if
+        else
+            System.Console.WriteLine(b)
+        end if
+    end sub
+end class")
+        End Function
+
+        <Fact>
+        Public Async Function NotMergedWithUnmatchingElseClauses2() As Task
+            ' Do not consider the using statement to be a simple block (as might be suggested by some language-agnostic helpers).
+            Await TestMissingInRegularAndScriptAsync(
+"class C
+    sub M(a as boolean, b as boolean)
+        if a then
+            [||]if b then
+                System.Console.WriteLine(a andalso b)
+            else
+                System.Console.WriteLine(a)
+            end if
+        else
+            using nothing
+                System.Console.WriteLine(a)
+            end using
+        end if
+    end sub
+end class")
+        End Function
+
+        <Fact>
+        Public Async Function NotMergedIntoElseIfWithUnmatchingElseIfClauses1() As Task
+            Await TestMissingInRegularAndScriptAsync(
+"class C
+    sub M(a as boolean, b as boolean)
+        if a orelse b then
+            System.Console.WriteLine()
+        elseif a then
+            [||]if b then
+                System.Console.WriteLine(a andalso b)
+            elseif a then
+                System.Console.WriteLine()
+            end if
+        elseif b then
+            System.Console.WriteLine()
+        end if
+    end sub
+end class")
+        End Function
+
+        <Fact>
+        Public Async Function NotMergedIntoElseIfWithUnmatchingElseIfClauses2() As Task
+            Await TestMissingInRegularAndScriptAsync(
+"class C
+    sub M(a as boolean, b as boolean)
+        if a orelse b then
+            System.Console.WriteLine()
+        elseif a then
+            [||]if b then
+                System.Console.WriteLine(a andalso b)
+            elseif a then
+                System.Console.WriteLine(a)
+            end if
+        elseif a then
+            System.Console.WriteLine(b)
+        end if
+    end sub
+end class")
+        End Function
+
+        <Fact>
+        Public Async Function NotMergedIntoElseIfWithUnmatchingElseClauses1() As Task
+            Await TestMissingInRegularAndScriptAsync(
+"class C
+    sub M(a as boolean, b as boolean)
+        if a orelse b then
+            System.Console.WriteLine()
+        elseif a then
+            [||]if b then
+                System.Console.WriteLine(a andalso b)
+            else
+                System.Console.WriteLine(a)
+            end if
+        else
+            System.Console.WriteLine(b)
+        end if
+    end sub
+end class")
+        End Function
+
+        <Fact>
+        Public Async Function NotMergedIntoElseIfWithUnmatchingElseClauses2() As Task
+            ' Do not consider the using statement to be a simple block (as might be suggested by some language-agnostic helpers).
```

Comment:
Do we need these tests, given the problem never would have hit in VB, and it might have been a misunderstanding from the start?

Reply:
It did hit in VB. This bug existed in my code for a while before I made this PR and I completely missed it. Even though this bug never shipped, it's a good regression test.

I don't understand what you mean by "do we need these tests". No tests are really "needed". But they're good to prevent bugs from creeping in.

Reply:
Yes, bugs are usually introduced by a misunderstanding, not intentionally.